src/app/lib/transaction.ts

```diff
@@ -79,6 +82,73 @@ export class OasisTransaction {
     return tw
   }
 
+  public static async buildStakingAllowTransfer(
+    nic: OasisClient,
+    signer: Signer,
+    to: string,
+    amount: bigint,
+  ): Promise<TW<oasis.types.StakingAllow>> {
+    const tw = oasis.staking.allowWrapper()
+    const nonce = await OasisTransaction.getNonce(nic, signer)
+    const beneficiary = await addressToPublicKey(to)
+
+    tw.setNonce(nonce)
+    tw.setFeeAmount(oasis.quantity.fromBigInt(0n))
+    tw.setBody({
+      beneficiary,
+      negative: false,
+      amount_change: oasis.quantity.fromBigInt(amount),
+    })
+
+    const gas = await tw.estimateGas(nic, signer.public())
+    tw.setFeeGas(gas)
+
+    return tw
+  }
+
+  public static async buildParatimeTransfer(
+    nic: OasisClient,
+    signer: Signer,
+    targetAddress: string,
+    fromAddress: string,
```

Comment:
`await publicKeyToAddress(signer.public())`  ?

Reply:
Probably

If that's correct, it can probably be simplified `await oasis.staking.addressFromBech32(await publicKeyToAddress(signer.public()))` to `await shortPublicKey(signer.public())`

Reply:
Does that work correctly for emerald withdrawals?

Reply:
It doesn't